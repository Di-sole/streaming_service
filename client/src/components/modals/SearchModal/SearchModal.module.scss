.container {
    justify-self: start;
    width: 100%;
    margin: 0 30px;
    max-width: 860px;
    padding-top: 60px;
    transition: 0.4s;

    &__title {
        font-size: 40px;
        line-height: 44px;
        font-weight: 900;
        color: white;
        margin-bottom: 40px;
    }

    &__content {
        padding: 24px 0 0 0;
        display: flex;
        flex-wrap: wrap;

        & a {
            margin-bottom: 10px;
            flex-basis: 50%;
            flex-shrink: 0;
        }
    }
}

@media (max-width: 1271px) {
    .container {
        width: 62%;
    }
}

@media (max-width: 1095px) {
    .container {
        max-width: none;
        width: 100%;
    }
}
