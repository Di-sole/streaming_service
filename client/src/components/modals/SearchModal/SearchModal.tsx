import ModalUI from '@/components/UI/Modal/ModalUI';
import styles from './SearchModal.module.scss';
import React, { useEffect, useMemo, useRef, useState } from 'react';
import ModalInputUI from '@/components/UI/ModalInput/ModalInputUI';
import SearchMovieLink from '@/components/UI/links/SearchMovieLink/SearchMovieLink';
import { useTranslation } from 'next-i18next';
import IMovies from '@/models/IMovies';
import { useRouter } from 'next/router';
import { GetStaticProps } from 'next';
import axios from 'axios';

const SearchModal = () => {
    const { t } = useTranslation('modals');
    const { locale } = useRouter();
    const [value, setValue] = useState<string>('');
    const [focus, setFocus] = useState<boolean>(false);
    const [searchMovies, setSearchMovies] = useState<IMovies[]>();
    const [isClose, setIsClose] = useState(false);
    const [movies, setMovies] = useState<IMovies[]>();

    useMemo(() => {
        const getMovies = async () => {
            try {
                const requestPerson = await axios.get(`http://localhost:6125/filmswithinfo`);
                setMovies(requestPerson.data);
            } catch (e: any) {
                console.log(e.response?.data?.message);
            }
        };
        getMovies();
    }, []);

    const inputWrapperRef = useRef(null);

    const clickHandler = (event: React.MouseEvent) => {
        const inputWrapper: HTMLDivElement = inputWrapperRef.current!;

        if (inputWrapper.firstElementChild === event.target) {
            setFocus(true);
        } else {
            setFocus(false);
        }
    };

    useEffect(() => {
        if (movies) {
            if (locale === 'ru') {
                setSearchMovies(
                    movies.filter((movie) => movie.name?.toLowerCase().includes(value.toLowerCase())).slice(0, 10),
                );
            } else {
                setSearchMovies(
                    movies.filter((movie) => movie.enName?.toLowerCase().includes(value.toLowerCase())).slice(0, 10),
                );
            }
        }
    }, [value]);

    return (
        <ModalUI close={isClose}>
            <div className={styles.container} onClick={(event) => clickHandler(event)}>
                <div className={styles.container__header}>
                    <h2 className={styles.container__title}>{t('searchModal.search')}</h2>

                    <div className={styles.container__inputWrapper} ref={inputWrapperRef}>
                        <ModalInputUI
                            focus={focus}
                            type="search"
                            inputType="string"
                            placeholder={t('searchModal.moviesSearch')}
                            value={value}
                            onChange={(value) => setValue(value)}
                            onClick={() => setValue('')}
                        />
                    </div>
                </div>

                <div className={styles.container__content}>
                    {searchMovies?.map((movie) => (
                        <SearchMovieLink
                            key={movie.id}
                            href={`/movies/${movie.id}`}
                            name={locale === 'ru' ? movie.name : movie.enName}
                            year={movie.year}
                            value={value}
                            onClick={() => setIsClose(true)}
                        />
                    ))}
                </div>
            </div>
        </ModalUI>
    );
};

export default SearchModal;
