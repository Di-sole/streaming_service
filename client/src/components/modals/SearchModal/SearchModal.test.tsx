import { render, screen } from '@testing-library/react';
import SearchModal from './SearchModal';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            asPath: '/',
        };
    },
}));

describe('SEARCHMODAL SNAPSHOTS TESTS', () => {
    test('SearchModal should not be changed', () => {
        render(<SearchModal />);

        const more = screen.getByTestId('searchModal');
        expect(more).toMatchSnapshot();
    });
});
