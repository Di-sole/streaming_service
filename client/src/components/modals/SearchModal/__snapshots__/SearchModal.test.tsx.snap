// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEARCHMODAL SNAPSHOTS TESTS SearchModal should not be changed 1`] = `
<div
  class="container"
  data-testid="searchModal"
>
  <div
    class="container__header"
  >
    <h2
      class="container__title"
    >
      searchModal.search
    </h2>
    <div
      class="container__inputWrapper"
    >
      <div
        class="container container_search "
      >
        <div
          class="container__placeholder"
        >
          searchModal.moviesSearch
        </div>
        <input
          class="container__input"
          type="string"
          value=""
        />
        <img
          alt="search"
          class="container__icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
    </div>
  </div>
  <div
    class="container__content"
  />
</div>
`;
